pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/ArgumentPatternMatcherTest.java

```diff
@@ -0,0 +1,36 @@
+package org.arquillian.algeron.pact.provider.core;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ArgumentPatternMatcherTest {
+
+    @Test
+    public void should_extract_arguments_if_number() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("I have (\\d+) cukes in my belly"), "I have 36 cukes in my belly");
+        assertThat(arguments).containsExactly("36");
+    }
+
+    @Test
+    public void should_extract_arguments_if_list() {
+        List<String> arguments = ArgumentPatternMatcher.arguments(Pattern.compile("The following animals: (.*)"), "The following animals: dog, cat, mouse");
+        assertThat(arguments).containsExactly("dog, cat, mouse");
```

Comment:
Well I think both is valid in this case since it is not a really big feature with thousands of classes related. For me in this case I see that one class extracts raw things, another one convert raw to type, and then there is the coordinator class that communicates with both objects.